src/AM.Condo/Tasks/GetLatestRelease.cs

```diff
@@ -0,0 +1,193 @@
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="GetLatestRelease.cs" company="automotiveMastermind and contributors">
+//   © automotiveMastermind and contributors. Licensed under MIT. See LICENSE and CREDITS for details.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace AM.Condo.Tasks
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using AM.Condo.Resources;
+    using Microsoft.Build.Framework;
+    using Microsoft.Build.Utilities;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Represents a Microsoft Build task used to locate and aquire the latest release of a GitHub project.
+    /// </summary>
+    public class GetLatestRelease : Task
+    {
+        #region Properties and Indexers
+        /// <summary>
+        /// Gets or sets the name of the organization that manages the project.
+        /// </summary>
+        [Required]
+        public string Organization { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the repository for which to get the release.
+        /// </summary>
+        [Required]
+        public string Repository { get; set; }
+
+        /// <summary>
+        /// Gets or sets the destination path where the release should be saved.
+        /// </summary>
+        [Required]
+        public string Destination { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the asset to retrieve.
+        /// </summary>
+        public string Asset { get; set; }
+
+        /// <summary>
+        /// Gets or sets the URI of the GitHub API.
+        /// </summary>
+        public string GitHubApi { get; set; } = "https://api.github.com";
+
+        /// <summary>
+        /// Gets or sets the authorization token used to authenticate to the GitHub API.
+        /// </summary>
+        public string Token { get; set; }
+        #endregion
+
+        #region Methods
+        /// <inheritdoc />
+        public override bool Execute()
+        {
+            // attempt to create the root API URL
+            if (!Uri.TryCreate(this.GitHubApi, UriKind.Absolute, out Uri baseUri))
```

Comment:
Has nothing to do with this, but curious on ideas on how to more elegantly handle TryParse/TryXYZ situations.